### PR TITLE
Codegen for compact Entity format

### LINKED ISSUE #138: Codegen for compact Entity format
I think it's better for us to preserve the functionality to generate the current long Entity format, and can be switched on with a command line arguemnt.

- [x] update codegen
- [x] update sea-orm-cli to generate short format by default
- [x] update docs

### PR DESCRIPTION
**Pull request by billy1624**
Resolve #138

### CODE REVIEW

**Comment by billy1624** [comment]
We should be able to generate correct `ForeignKeyAction` for each relation given that we have it now.

**Comment by billy1624** [comment]
Hey @tyt2y3, see the generated compact entity files here bd6e5f8

**Comment by tyt2y3** [comment]
Can't comment there. Why the contact_details is string instead of json?

**Comment by billy1624** [comment]
> Can't comment there. Why the contact_details is string instead of json?

See it here #146 

Fixed the json thing on SeaQL/sea-schema@e7f6a37

**Comment by tyt2y3** [comment]
How about the `Decimal` type?

```
#[sea_orm(column_type = "Decimal(Some((19, 4)))")]
```

without this, the ColumnDef won't compile

**Comment by billy1624** [comment]
Can we provide a default column type for Decimal?

**Comment by billy1624** [comment]
Also, should we fully quantify custom types? E.g. Decimal, Json

**Comment by tyt2y3** [comment]
The problem is the Decimal type does not work without specifying the decimal places

**Comment by tyt2y3** [comment]
> Also, should we fully quantify custom types? E.g. Decimal, Json

Basically yes

**Comment by billy1624** [comment]
> The problem is the Decimal type does not work without specifying the decimal places

We can provide a default for it if it can't be discovered? Let say Decimal((10,2))

**Comment by tyt2y3** [comment]
> > The problem is the Decimal type does not work without specifying the decimal places
> 
> We can provide a default for it if it can't be discovered? Let say Decimal((10,2)) as default?

I'd rather let it not compile

**Comment by billy1624** [comment]
> Looks good to me! Anything missing?

Docs to Add / Edit

1. Document `compact` / `expanded` options for `sea-orm-cli generate entity`
2. Introduce compact entity syntax & `DeriveEntityModel` macro attributes

File: sea-orm-codegen/src/entity/writer.rs [hunk 1]
```diff
@@ -297,25 +320,83 @@ impl EntityWriter {
...
+            .iter()
+            .map(|col| {
+                let mut attrs: Punctuated<_, Comma> = Punctuated::new();
+                if primary_keys.contains(&col.name) {
+                    attrs.push(quote! { primary_key });
+                    if !col.auto_increment {
+                        attrs.push(quote! { auto_increment = false });
+                    }
+                }
+                if let Some(ts) = col.get_col_type_attrs() {
```
Comment (tyt2y3) [inline sea-orm-codegen/src/entity/writer.rs:344]: We forgot one thing. That is, `nullable` when a col_type is involved.
See https://github.com/SeaQL/sea-orm/issues/105#issuecomment-915922635

Comment (billy1624) [reply sea-orm-codegen/src/entity/writer.rs:344]: Oh, will fix this

**Pull request merged**